lib/ansible/module_utils/common/warnings.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+global_warnings = []
+
+
+class AnsibleWarning(Warning):
```

Comment:
we also have a 'debug' facility that works the same, should we just add it to this PR?

Reply:
Might not be a bad idea. I'm not familiar with that mechanism.

Reply:
After working through this more, I don't think moving `debug` out would make sense with this change. That is more a flag rather than a container for holding values. We could handle that in other PR maybe, but it seems out of context for this change.

Reply:
current debug just pushes to local syslog, but with this functioning the same as 'warnings' and return accumulated messages back to controller